Core/Lib/API/APIModel.php

```diff
@@ -217,10 +224,40 @@ public function doPOST(): bool
             $this->params[] = $message['message'];
         }
 
-        $this->setError('bad-request', $values);
+        $this->setError('bad-request', $this->request->request->all());
         return false;
     }
 
+    /**
+     * Process the POST (create) request. Overwrite this function to implement is functionality.
+     *
+     * @return bool
+     */
+    public function doPOST(): bool
```

Comment:
doPost and doPut is almost the same code, just for the condition of exist or not the resource. Probably it is better refactor in a private method which is called by doPost or doPut to avoid code duplication